### PR TITLE
bump to go 1.22

### PR DESCRIPTION
**Pull request by OliverMKing**
# Description

bumps us to Go 1.22. Needed for a Gatekeeper integration for scanning our manifests.

Also update a few unit tests to reflect updated controller runtime packages. Fake client no longer fully infers types.

### CODE REVIEW

**Review by tbarnes94 — APPROVED** [review]

File: pkg/controller/service/ingress_reconciler_test.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ func TestIngressReconcilerIntegration(t *testing.T) {
 	svc := &corev1.Service{}
 	svc.UID = "test-svc-uid"
 	svc.Name = "test-service"
+	svc.APIVersion = "v1"
+	svc.Kind = "Service"
```
Comment (Vidya2606) [inline pkg/controller/service/ingress_reconciler_test.go:39]: for my understanding: what is the use of adding apiversion and kind here ?

Comment (OliverMKing) [reply pkg/controller/service/ingress_reconciler_test.go:39]: It's just a difference in how the fake client works due to the version bump.

**Review by bfoley13 — APPROVED** [review]

**Pull request merged**